### ISSUE TITLE
DNS Queries still happening on blocked domains

### ISSUE THREAD

**Issue by cbuijs** [uBlock-LLC/uBlock#232]
Looks like that the domains blocked on a web-page are still resolved in DNS by the browser. Can this be prevented somehow? It is overhead that maybe can be omitted and speedup things.

**Comment by gorhill** [comment]
Could this be related to the setting _"Predict network actions to improve page load performance"_? uBlock uses [webRequest.onBeforeRequest](https://developer.chrome.com/extensions/webRequest), there is nothing earlier than this.

**Comment by cbuijs** [comment]
Indeed! Good catch! Switched it of and the DNS stays clean of blocked domains. I cannot measure if it is quicker, but at least it is less overhead :-).

**Comment by gorhill** [comment]
Added suggestion to turn off the setting to the [_"Privacy stuff"_ wiki page](https://github.com/gorhill/uBlock/wiki/Privacy-stuff).

**Comment by ghost** [comment]
EDIT: https://code.google.com/p/chromium/issues/detail?id=415071

I have today compiled Chromium from the repository, current version
Version 39.0.2165.0 (64-bit)
and I can confirm that when
`Predict network actions to improve page load performance` is selected
then uBlock and HTTPSwitchboard and HTTPS Everywhere  extension work as expected.
However when it's deselected, then none of them work at all, they are unexpectedly broken as if javascript wasn't running within them, or something.
I can reproduce this at will.
Unsure if I should make a new issue or just go to chromium-dev google groups. I have only just minimized the issue to that option being deselected.

and there are no reported errors in chrome://extensions
some screens:
![predict1](https://cloud.githubusercontent.com/assets/5661825/4367398/46a457dc-42d6-11e4-804b-20ce8cf22daf.png)
![predict2](https://cloud.githubusercontent.com/assets/5661825/4367399/497c34ac-42d6-11e4-9a5f-ca6600b35cb3.png)
![predict3](https://cloud.githubusercontent.com/assets/5661825/4367401/4c5cb4bc-42d6-11e4-8230-8548ab143373.png)
![predict4](https://cloud.githubusercontent.com/assets/5661825/4367402/4ed5b4aa-42d6-11e4-814f-8f715e6adfcc.png)

**Comment by gorhill** [comment]
I can't support unstable version of Chromium, I just don't have the time/resource for this. In the current case the problem is obviously with Chromium, so this should be reported to Chromium devs.
